src/SdkCommon/ClientRuntime/ClientRuntime/RetryAfterDelegatingHandler.cs

```diff
@@ -54,15 +54,16 @@ public RetryAfterDelegatingHandler(HttpMessageHandler innerHandler)
             CancellationToken cancellationToken)
         {
             HttpResponseMessage previousResponseMessage = null;
-            do {
+            do
+            {
                 HttpResponseMessage response = null;
 
                 try
                 {
                     response = await base.SendAsync(request, cancellationToken).ConfigureAwait(false);
 
-                    // if they send back a 429 and there is a retry-after header
-                    if (response.StatusCode == (HttpStatusCode)429 && response.Headers.Contains("Retry-After"))
+                    // if they send back a 429 or a 503
+                    if (response.StatusCode == ((HttpStatusCode)429) || response.StatusCode == HttpStatusCode.ServiceUnavailable)
```

Comment:
@joerage Retry-After is meant to handle 429 when Retry-After headers are sent by the service.
Retry-After has a very specific needs for throttling for other RPs and that behavior should not be changed.
Is there a scenario where you think Retry-After header will be used for 503 which is not covered by default retry policy that currently handles 503 scenarios?

Reply:
From ARM specs:
The retry interval, strategy (e.g. exponential/linear) and max timeout should be defined by the server and not the client. The server indicates how long it expects the client to wait before retrying the call by including a (Retry-After header)

That does not only applies to the 429 responses, but for ServiceUnavailable as well.

Reply:
@joerage let me make sure I understand this scenario, are you saying when your service (DTL) is unavailable (503) are you saying Retry-After header will be sent? If yes who is sending that header, is it DTL or ARM?
Because the change you have added is suggesting, we should be honoring Retry-After during 503.

Reply:
DTL is calling other RPs/services. If those RP responds with a 503, and also add a Retry-After header, we should honor that value before trying again. The default with the SDK at the moment is depending on the client retry strategy, which could come into conflict with what the RP/service is telling us to do.

Reply:
@joerage will you be able to give me a scenario or any network traces that shows you get Retry-After during 503
I am not sure if a service is involved when an user get's 503, if a service is unavailable how can they add Retry-After header to the response?
Maybe I am failing to understand your scenario here.

Reply:
Yes, when a 503 is returned, the server received the request and returned that error code as it is unable to handle the request. 
From the HTTP specs:
The server is currently unable to handle the request due to a temporary overloading or maintenance of the server. The implication is that this is a temporary condition which will be alleviated after some delay. If known, the length of the delay MAY be indicated in a Retry-After header. If no Retry-After is given, the client SHOULD handle the response as it would for a 500 response.